pkg/server/plugin/keymanager/hashicorpvault/hashicorp_vault.go

```diff
@@ -0,0 +1,675 @@
+	SecretID string `hcl:"approle_secret_id" json:"approle_secret_id"`
+}
+
+// CertAuthConfig represents parameters for cert auth method
+type CertAuthConfig struct {
+	// Name of the mount point where Client Certificate Auth method is mounted. (e.g., /auth/<mount_point>/login)
+	// If the value is empty, use default mount point (/auth/cert)
+	CertAuthMountPoint string `hcl:"cert_auth_mount_point" json:"cert_auth_mount_point"`
+	// Name of the Vault role.
+	// If given, the plugin authenticates against only the named role.
+	CertAuthRoleName string `hcl:"cert_auth_role_name" json:"cert_auth_role_name"`
+	// Path to a client certificate file.
+	// Only PEM format is supported.
+	ClientCertPath string `hcl:"client_cert_path" json:"client_cert_path"`
+	// Path to a client private key file.
+	// Only PEM format is supported.
+	ClientKeyPath string `hcl:"client_key_path" json:"client_key_path"`
+}
+
+// K8sAuthConfig represents parameters for Kubernetes auth method.
+type K8sAuthConfig struct {
+	// Name of the mount point where Kubernetes auth method is mounted. (e.g., /auth/<mount_point>/login)
+	// If the value is empty, use default mount point (/auth/kubernetes)
+	K8sAuthMountPoint string `hcl:"k8s_auth_mount_point" json:"k8s_auth_mount_point"`
+	// Name of the Vault role.
+	// The plugin authenticates against the named role.
+	K8sAuthRoleName string `hcl:"k8s_auth_role_name" json:"k8s_auth_role_name"`
+	// Path to the Kubernetes Service Account Token to use authentication with the Vault.
+	TokenPath string `hcl:"token_path" json:"token_path"`
+}
+
+// Plugin is the main representation of this keymanager plugin
+type Plugin struct {
+	keymanagerv1.UnsafeKeyManagerServer
+	configv1.UnsafeConfigServer
+
+	logger     hclog.Logger
+	serverID   string
+	mu         sync.RWMutex
+	entries    map[string]keyEntry
+	entriesMtx sync.RWMutex
+
+	authMethod AuthMethod
+	cc         *ClientConfig
+	vc         *Client
+
+	scheduleDelete chan string
+	cancelTasks    context.CancelFunc
+
+	hooks pluginHooks
+}
+
+// New returns an instantiated plugin.
+func New() *Plugin {
+	return newPlugin()
+}
+
+// newPlugin returns a new plugin instance.
+func newPlugin() *Plugin {
+	return &Plugin{
+		entries: make(map[string]keyEntry),
+		hooks: pluginHooks{
+			lookupEnv: os.LookupEnv,
+			clk:       clock.New(),
+		},
+		scheduleDelete: make(chan string, 120),
+	}
+}
+
+// SetLogger sets a logger
+func (p *Plugin) SetLogger(log hclog.Logger) {
+	p.logger = log
+}
+
+func (p *Plugin) Configure(ctx context.Context, req *configv1.ConfigureRequest) (*configv1.ConfigureResponse, error) {
+	config := new(Config)
+
+	if err := hcl.Decode(&config, req.HclConfiguration); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "unable to decode configuration: %v", err)
+	}
+
+	serverID := config.KeyIdentifierValue
+	if serverID == "" {
+		var err error
+
+		if serverID, err = getOrCreateServerID(config.KeyIdentifierFile); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
We need to validate that the configuration has a key identifier file or a key identifier value, and that the configuration does not have a key identifier file and a key identifier value at the same time. You can look at how this is validated in other key managers that have those settings.